src/Core/Plugin.php

```diff
@@ -80,6 +80,20 @@ public static function configPath(string $name): string
         return $plugin->getConfigPath();
     }
 
+    /**
+     * Returns the filesystem path for plugin's folder containing template files.
+     *
+     * @param string $name name of the plugin in CamelCase format.
+     * @return string Path to the plugin folder container config files.
+     * @throws \Cake\Core\Exception\MissingPluginException If plugin has not been loaded.
+     */
+    public static function templatePath(string $name): string
```

Comment:
why we need static method in `Plugin` class ?
this is useless IMO

Reply:
Static methods are required in `App::path()` and `ExtractTask` to get plugin template path. Without this new method you would still have to use static method `Plugin::path()` and string concatenation.

Reply:
you can use 
```php
Plugin::getCollection()->get('PluginName')->getTemplatePath()
```
instead of `templatePath()` we can have a new method `get()` and then 
```php
Plugin::get('PluginName')->getTemplatePath()
```

Reply:
I added `Plugin::templatePath()` for consistency. If / when we decide to get rid of other methods like `Plugin::configPath()`, `Plugin::classPath()` when can remove `Plugin::templatePath()` too.